### PR TITLE
introduce end to end tests

### PR DESCRIPTION
**Pull request by rojkov**
This is preliminary e2e tests.

All test cases run in isolation: for each case a brand new resource manager is created and started together with a fake CRI gRPC server.

I'm going to simplify the code structure to make it more readable, hence WIP.

### CODE REVIEW

**Comment by klihub** [comment]
This looks interesting !

**Comment by rojkov** [comment]
I'm kinda satisfied now with the code. The PR is ready for review.

File: test/functional/e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+)
+
+const (
+	testDir = "/tmp/cri-rm-test"
+)
+
+func setUp(t *testing.T) {
+	t.Helper()
+
+	if err := os.MkdirAll(testDir, 0700); err != nil {
```
Comment (marquiz) [inline test/functional/e2e_test.go]: Should we use `ioutil.TempDir` or smth to avoid possible races?

Comment (rojkov) [reply test/functional/e2e_test.go]: Agree. Even though mkdir is an atomic syscall there's no need to call it for every test function. I put it to `init()`.

**Pull request merged**